src/plugins/dashboard/public/application/lib/save_dashboard.ts

```diff
@@ -42,24 +43,18 @@ import { DashboardStateManager } from '../dashboard_state_manager';
  * dashboard.
  */
 export function saveDashboard(
-  toJson: (obj: any) => string,
   timeFilter: TimefilterContract,
-  dashboardStateManager: DashboardStateManager,
+  stateContainer: DashboardAppStateContainer,
+  savedDashboard: any,
   saveOptions: SavedObjectSaveOpts
 ): Promise<string> {
-  const savedDashboard = dashboardStateManager.savedDashboard;
-  const appState = dashboardStateManager.appState;
+  const appState = stateContainer.getState();
 
-  updateSavedDashboard(savedDashboard, appState, timeFilter, toJson);
+  updateSavedDashboard(savedDashboard, appState, timeFilter);
 
   return savedDashboard.save(saveOptions).then((id: string) => {
     if (id) {
-      // reset state only when save() was successful
```

Comment:
is there a todo here?

Reply:
Yea will add a task to investigate if resetState() is needed here when saving the dashboard.

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4003